### PR TITLE
add set version script

### PR DESCRIPTION
**Pull request by ShiCheng-Lu**
There are two components of this PR, improved CI checks, and version updating script.

1. Improved CI checks.
- adding a check to make sure both versions do not exist in pod yet
- lint both SDKs before uploading either
This should make sure that if CI passes, both are successfully uploaded, and if CI fails, neither are uploaded.

2. version updating script.
- this script does everthing in the iOS version bump/release document in notion, I believe running the script will overall result in less version bumping mistakes than doing it by hand every time. Of course it's still recommended to follow through the document and check the changes in git.

(There was no documentation on bumping MotionSDK in notion, which should have been there, I've updated the notion document)

### CODE REVIEW

File: .github/workflows/release-sdk.yml [hunk 1]
```diff
@@ -74,12 +74,21 @@ jobs:
...
+          echo "RadarSDKMotion version is OK"
+
       - name: Deploy to Cocoapods
         run: |
           set -eo pipefail
           pod lib lint --allow-warnings RadarSDK.podspec
-          pod trunk push --allow-warnings RadarSDK.podspec
           pod lib lint --allow-warnings RadarSDKMotion.podspec
+          pod trunk push --allow-warnings RadarSDK.podspec
           pod trunk push --allow-warnings RadarSDKMotion.podspec
```
Comment (ShiCheng-Lu) [inline .github/workflows/release-sdk.yml:92]: we definitely want to lint both before publishing any of the two.

File: RadarSDK.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -949,7 +949,6 @@
 					"@executable_path/Frameworks",
 					"@loader_path/Frameworks",
 				);
-				MARKETING_VERSION = 3.9.6;
```
Comment (ShiCheng-Lu) [inline RadarSDK.xcodeproj/project.pbxproj:952]: this is the package settings, which overwrites the project settings, we should just have the version set for project settings which will be resolved everywhere to the single correct marketing version

File: RadarSDK.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -979,7 +978,6 @@
...
```
Comment (ShiCheng-Lu) [inline RadarSDK.xcodeproj/project.pbxproj:982]: same as above

File: RadarSDKMotion/RadarSDKMotion.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -347,6 +348,7 @@
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				LOCALIZATION_PREFERS_STRING_CATALOGS = YES;
+				MARKETING_VERSION = 3.18.4;
```
Comment (ShiCheng-Lu) [inline RadarSDKMotion/RadarSDKMotion.xcodeproj/project.pbxproj:351]: adding the marketing version to motion sdks. (project setting)

File: set_version.sh
Comment (lmeier) [inline set_version.sh:1]: I think there's also a `CFBundleShortVersionString` in the `Info.plist` that has lagged for a while and we don't generally update. We should get that too.

**Review by lmeier — COMMENTED** [review]
I'm gonna test this with a bump today to verify it's good before 👍, but this looks good. Thank you — nice proactive change.

File: set_version.sh
Comment (ShiCheng-Lu) [reply set_version.sh:1]: CFBundleShortVersionString is set to parameter substitution of #(MARKETING_VERSION) so we are good there.

**Review by KennyHuRadar — APPROVED** [review]
script works well from my testing and the changes to release script makes sense imo

**Pull request merged**